lib/core/topologies/read_preference.js

```diff
@@ -100,20 +100,23 @@ const VALID_MODES = [
  * @param {object} options The options object from which to extract the read preference.
  * @return {ReadPreference}
  */
-ReadPreference.fromOptions = function(options) {
-  if (!options) return null;
+ReadPreference.fromOptions = options => {
+  if (!options) return;
   const readPreference = options.readPreference;
-  if (!readPreference) return null;
   const readPreferenceTags = options.readPreferenceTags;
-  const maxStalenessSeconds = options.maxStalenessSeconds;
+
+  if (readPreference == null) {
+    return;
+  }
+
   if (typeof readPreference === 'string') {
     return new ReadPreference(readPreference, readPreferenceTags);
   } else if (!(readPreference instanceof ReadPreference) && typeof readPreference === 'object') {
     const mode = readPreference.mode || readPreference.preference;
     if (mode && typeof mode === 'string') {
       return new ReadPreference(mode, readPreference.tags, {
-        maxStalenessSeconds: readPreference.maxStalenessSeconds || maxStalenessSeconds,
-        hedge: readPreference.hedge
+        maxStalenessSeconds: readPreference.maxStalenessSeconds,
+        hedge: options.hedge
```

Comment:
what motivated these changes? It looks like we've lost support for `maxStalenessSeconds` from `options`, and I believe `hedge` is not a valid connection string option

Reply:
yes your right, this is in draft now.

Reply:
I pulled changes from master thinking that had the updated code, seems like master is wrong and needs updates reflected in 3.6

Reply:
Will pull in these changes in master